lib/src/model/container.dart

```diff
@@ -5,121 +5,215 @@
 import 'package:analyzer/dart/element/element.dart';
 import 'package:dartdoc/src/model/model.dart';
 import 'package:dartdoc/src/model_utils.dart' as model_utils;
+import 'package:meta/meta.dart';
 import 'package:quiver/iterables.dart' as quiver;
 
-// Can be either a Class or Extension, used in the package graph and template data.
-// Aggregates some of the common getters.
+/// A [Container] represents a Dart construct that can contain methods,
+/// operators, and fields, such as [Class], [Enum], or [Extension].
+///
+/// Member naming in [Container] follows these general rules:
+///
+/// **instance** : Members named 'instance' contain the children of this
+/// container that can be referenced from within the container without a prefix.
+/// Usually overridden in subclasses with calls to super.
+/// **constant** : Members named 'constant' contain children declared constant.
+/// **variable** : The opposite of constant.  For the templating system.
+/// **static** : Members named 'static' are related to static children of this
+/// container.
+/// **public** : Filtered versions of the getters showing only public items.
+/// Mostly for the templating system.
+/// **sorted** : Filtered versions of the getters creating a sorted list by
+/// name.  For the templating system.
+/// **has** : boolean getters indicating whether the underlying getters are
+/// empty.  Mostly for the templating system.
+/// **all** : Referring to all children.
 abstract class Container extends ModelElement {
-  List<Field> _constants;
-  List<Operator> _operators;
-  List<Method> _staticMethods;
-  List<Method> _instanceMethods;
-  List<Field> _staticFields;
-  List<Field> _instanceFields;
-
   Container(Element element, Library library, PackageGraph packageGraph)
       : super(element, library, packageGraph, null);
 
   bool get isClass => element is ClassElement;
-
   bool get isExtension => element is ExtensionElement;
 
-  List<Method> get methods => [];
+  @mustCallSuper
+  Iterable<ModelElement> get allModelElements => quiver.concat([
```

Comment:
Hmm, should we just use spread here? Does quiver.concat buy us something in a getter?